pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -248,9 +243,65 @@ func newDeploymentSettingsInitCmd() *cobra.Command {
 		&stack, "stack", "s", "",
 		"The name of the stack to operate on. Defaults to the current stack")
 
+	cmd.PersistentFlags().BoolVarP(
+		&force, "force", "f", false,
+		"Forces content to be generated even if it is already configured")
+
 	return cmd
 }
 
+func initStackDeploymentCmd(d *deploymentSettingsCommandDependencies, gitSSHPrivateKeyPath string) error {
+	d.Deployment = &workspace.ProjectStackDeployment{
+		DeploymentSettings: apitype.DeploymentSettings{},
+	}
+
+	err := configureGit(d, gitSSHPrivateKeyPath)
+	if err != nil {
+		return err
+	}
+
+	err = configureImageRepository(d)
+	if err != nil {
+		return err
+	}
+
+	err = configureAdvancedSettings(d)
+	if err != nil {
+		return err
+	}
+
+	option := promptUserSkippable(
+		d.Yes,
+		"Do you want to configure an OpenID Connect integration?",
+		[]string{
+			optNo,
+			optOidcAws,
+			optOidcAzure,
+			optOidcGcp,
+		},
+		optNo,
+		d.DisplayOptions.Color)
+
+	switch option {
+	case optOidcAws:
+		err = configureOidcAws(d)
+	case optOidcAzure:
+		err = configureOidcAzure(d)
+	case optOidcGcp:
+		err = configureOidcGCP(d)
+	}
+	if err != nil {
+		return err
+	}
+
+	err = saveProjectStackDeployment(d.Deployment, d.Stack)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
Any reason not to just inline this? I feel like if we want to pull it out we should at least document the function.

Reply:
just leftover of moving code around, will change

Reply:
done